ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Storage/LedgerDB/Snapshots.hs

```diff
@@ -161,15 +163,25 @@ data DiskSnapshot = DiskSnapshot {
       -- snapshot number matching the slot number of the corresponding ledger
       -- state. We only use the snapshots numbers to determine the order in
       -- which we try them.
-      dsNumber :: Word64
+      dsNumber   :: Word64
 
       -- | Snapshots can optionally have a suffix, separated by the snapshot
       -- number with an underscore, e.g., @4492799_last_Byron@. This suffix acts
       -- as metadata for the operator of the node. Snapshots with a suffix will
       -- /not be trimmed/.
-    , dsSuffix :: Maybe String
+    , dsSuffix   :: Maybe String
+
+    -- | Snapshots can carry a checksum, which is filled in once they
+    --   have been written to disk. When reading a snapshot, the checksum
+    --   of the read data will be checked to match to detect data corruption.
+    , dsChecksum :: Maybe CRC
```

Comment:
My understanding is that `DiskSnapshot` represents "the name of the snapshot" and this CRC field represents the contents. I think that there should be a `DiskSnapshot` datatype and on top a:

```
data DiskSnapshotWithCRC = DiskSnapshotWithCRC {
  snapshotName :: !DiskSnapshot
  snapshotCRC :: !CRC
}
```
And whenever we have one of those, the CRC must be present (not a `Maybe`).